conans/client/conf/__init__.py

```diff
@@ -234,14 +235,22 @@ def storage_path(self):
             conan_user_home = os.getenv("CONAN_USER_HOME")
             if conan_user_home:
                 storage = self.storage["path"]
+                # The ~ in the storage means relative to conan user home, too late to change
+                # this mess
                 if storage[:2] == "~/":
                     storage = storage[2:]
                 result = os.path.join(conan_user_home, storage)
             else:
                 result = conan_expand_user(self.storage["path"])
+                if not os.path.isabs(result):
+                    raise ConanException("Specify an absolute path in the 'path' variable of "
+                                         "the conan.conf file")
         except KeyError:
             result = None
-        result = get_env('CONAN_STORAGE_PATH', result)
+        result = conan_expand_user(get_env('CONAN_STORAGE_PATH', result))
```

Comment:
This will raise an Exception when CONAN_STORATE_PATH is not defined and result is None. 
Instead of returning None, previous behavior